### PR TITLE
wireguard-exporter: fix static build

### PR DESCRIPTION
**Pull request by VergeDX**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/monitoring/prometheus/wireguard-exporter/static-build-fix.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+--- a/.cargo/config.toml
++++ b/.cargo/config.toml
+@@ -1,8 +1,5 @@
+ # This is used in the Docker build, you might need to adjust it for local usage.
+ 
+-[target.x86_64-unknown-linux-musl]
+-linker = "x86_64-linux-musl-gcc"
+-
+ [target.aarch64-unknown-linux-musl]
+ linker = "aarch64-linux-musl-gcc"
```
Comment (NickCao) [inline pkgs/servers/monitoring/prometheus/wireguard-exporter/static-build-fix.patch:12]: What about other platform? Shall we just delete this file altogether.

File: pkgs/servers/monitoring/prometheus/wireguard-exporter.nix [hunk 2]
```diff
@@ -13,6 +13,8 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "sha256-NsxGpjuZPpz4gCJRp5IOcfRFh8DTud47nV2bE0/kc2Q=";
 
+  patchPhase = lib.optionals stdenv.targetPlatform.isStatic "rm .cargo/config.toml";
```
Comment (NickCao) [inline pkgs/servers/monitoring/prometheus/wireguard-exporter.nix:16]: Do this unconditionally is more preferable.

**Comment by NickCao** [comment]
Also the commit message should be: `wireguard-exporter: fix static build`

File: pkgs/servers/monitoring/prometheus/wireguard-exporter.nix [hunk 3]
```diff
@@ -13,6 +13,8 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "sha256-NsxGpjuZPpz4gCJRp5IOcfRFh8DTud47nV2bE0/kc2Q=";
 
+  patchPhase = "rm .cargo/config.toml";
```
Comment (NickCao) [inline pkgs/servers/monitoring/prometheus/wireguard-exporter.nix:16]:
```suggestion
  postPatch = ''
    # drop hardcoded linker names, fixing static build
    rm .cargo/config.toml
  '';
```

File: pkgs/servers/monitoring/prometheus/wireguard-exporter.nix [hunk 4]
```diff
@@ -13,6 +13,11 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "sha256-NsxGpjuZPpz4gCJRp5IOcfRFh8DTud47nV2bE0/kc2Q=";
 
+  patchPhase = ''
```
Comment (NickCao) [inline pkgs/servers/monitoring/prometheus/wireguard-exporter.nix:16]: Please use `postPatch` instead of `patchPhase`, otherwise `patches` are ignored.

**Review by Ma27 — APPROVED** [review]

**Pull request merged**